### PR TITLE
Enable building clippy in CI

### PR DESCRIPTION
**Pull request by oli-obk**
r? @alexcrichton 

As discussed at Rustfest. Measured additional time is 4 minutes on my machine if no dependencies are shared with other tools. In reality most dependencies are shared (especially the slow to compile ones like serde).

cc @Manishearth 

Does not run clippy's test suite, since

a) it is nontrivial in the rustc build system
b) it breaks more frequently but the breakage is negligible

If clippy breaks, the procedure to follow is documented under https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#external-dependencies

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

:+1:

**Comment by Manishearth** [comment]
We should also ask folks to ping you/me/llogiq for clippy and you/scott for miri in the document? That way we can keep an eye out and help.

File: src/tools/toolstate.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 miri = "Broken"
 
 # ping @Manishearth @llogiq @mcarton @oli-obk
-clippy = "Broken"
+clippy = "Building"
```
Comment (MaloJaffre) [inline src/tools/toolstate.toml]: This seems to break CI:
```
[00:02:42] failed to parse TOML configuration 'toolstate.toml': unknown variant `Building`, expected one of `Compiling`, `Testing`, `Broken` for key `clippy`

[00:02:42] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap build nonexistent/path/to/trigger/cargo/metadata

[00:02:42] Build completed unsuccessfully in 0:01:16

[00:02:42] Makefile:79: recipe for target 'prepare' failed

[00:02:42] make: *** [prepare] Error 1

[00:02:42] Command failed. Attempt 2/5:

[00:02:42]     Finished dev [unoptimized] target(s) in 0.0 secs
```

Comment (oli-obk) [reply src/tools/toolstate.toml]: Darn it... last minute changes are never a good idea... sorry about that. I also fixed the wrong docs on the top of the file

**Comment by oli-obk** [comment]
@Manishearth the pinging is already suggested in toolstate.toml

**Pull request merged**